src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -177,9 +177,26 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     bindingContext.ModelState.AddModelError(modelBindingKey, message);
                 }
             }
-            catch (Exception ex)
+            catch (InputFormatterException InputFormatterException)
             {
-                bindingContext.ModelState.AddModelError(modelBindingKey, ex, bindingContext.ModelMetadata);
+                bindingContext.ModelState.AddModelError(modelBindingKey, InputFormatterException, bindingContext.ModelMetadata);
+            }
+            catch (Exception exception)
```

Comment:
You could use exception filters to avoid unwinding the stack when you're going to no-op:

```
catch (Exception ex) when (ShouldHandleException())
{
    ...
}

bool ShouldHandleException()
{
    if (formatter is IFormatterExceptionPolicy exceptionPolicy && exceptionPolicy.HandlesExceptions)
    {
        return _options.ConvertAllExceptionsOfInbuiltInputFormattersToModelStateErrors;
    }

    return _options.ConvertAllExceptionsToModelStateErrors;
}
```